src/PKSim.Core/Model/IndividualProteinTask.cs

```diff
@@ -0,0 +1,273 @@
+            CanBeVaried = true,
+            CanBeVariedInPopulation = true,
+            Dimension = MOLAR_CONCENTRATION,
+            GroupName = CoreConstants.Groups.RELATIVE_EXPRESSION,
+         };
+
+      protected IParameter createFormulaParameterIn(IContainer parameterContainer, ParameterRateMetaData parameterRateMetaData, string moleculeName,
+         string groupName = null)
+      {
+         var parameter = _parameterFactory.CreateFor(parameterRateMetaData, new FormulaCache());
+         parameterContainer.Add(parameter);
+         if (parameterRateMetaData.DefaultUnit != null)
+            parameter.DisplayUnit = parameter.Dimension.Unit(parameterRateMetaData.DefaultUnit);
+
+         if (!string.IsNullOrEmpty(groupName))
+            parameter.GroupName = groupName;
+
+         parameter.Formula.ReplaceKeywordsInObjectPaths(new[] {ObjectPathKeywords.MOLECULE}, new[] {moleculeName});
+         return parameter;
+      }
+
+      public override IndividualMolecule AddMoleculeTo(ISimulationSubject simulationSubject, string moleculeName)
+      {
+         var globalContainer = CreateMolecule(moleculeName);
+         AddVascularSystemExpressionNew(globalContainer, CoreConstants.Groups.VASCULAR_SYSTEM,
+            relExpParam(REL_EXP_BLOOD_CELLS),
+            fractionParam(FRACTION_EXPRESSED_BLOOD_CELLS, CoreConstants.Rate.ZERO_RATE),
+            fractionParam(FRACTION_EXPRESSED_BLOOD_CELLS_MEMBRANE, CoreConstants.Rate.PARAM_F_EXP_BC_MEMBRANE, editable: false)
+         );
+
+         AddVascularSystemExpressionNew(globalContainer, CoreConstants.Groups.VASCULAR_SYSTEM, relExpParam(REL_EXP_PLASMA));
+
+         AddVascularSystemExpressionNew(globalContainer, CoreConstants.Groups.VASCULAR_SYSTEM,
+            relExpParam(REL_EXP_VASC_ENDO),
+            fractionParam(FRACTION_EXPRESSED_VASC_ENDO_ENDOSOME, CoreConstants.Rate.ZERO_RATE),
+            fractionParam(FRACTION_EXPRESSED_VASC_ENDO_APICAL, CoreConstants.Rate.ZERO_RATE),
+            fractionParam(FRACTION_EXPRESSED_VASC_ENDO_BASOLATERAL, CoreConstants.Rate.PARAM_F_EXP_VASC_BASOLATERAL, editable: false)
+         );
+
+         AddTissueOrgansExpressionNew(simulationSubject, moleculeName);
+         AddLumenExpressionsNew(simulationSubject, moleculeName);
+         AddMucosaExpressionNew(simulationSubject, moleculeName);
+
+         simulationSubject.AddMolecule(globalContainer);
+
+         _individualPathWithRootExpander.AddRootToPathIn(simulationSubject, moleculeName);
+         return globalContainer;
+      }
+
+      protected void AddMucosaExpressionNew(ISimulationSubject simulationSubject, string moleculeName)
+      {
+         foreach (var organ in simulationSubject.Organism.OrgansByName(CoreConstants.Organ.SmallIntestine, CoreConstants.Organ.LargeIntestine))
+         {
+            var organMucosa = organ.Compartment(CoreConstants.Compartment.Mucosa);
+            foreach (var compartment in organMucosa.GetChildren<Compartment>().Where(c => c.Visible))
```

Comment:
do we have non-visible mucosa compartments? or why is `.Where(c => c.Visible)` required?

Reply:
this is the old implementation. no idea